substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/DebugHelper.java

```diff
@@ -281,6 +295,493 @@ public static void printLocationInfo(@SuppressWarnings("unused") IsolateThread t
+                            .string(": ")
+                            .unsigned(frameInfo.getSourceLineNumber())
+                            .string(")")
+                            .newline();
+                    return false;
+                }
+            });
+
+            return Word.objectToUntrackedPointer(classLoaders[0]);
+        }
+
+        /**
+         * Returns loaded class by name for given classloader.
+         * @param thread
+         * @param ptrClassLoader classloader which loaded named class.
+         * @param cClassName name of desired class.
+         * @return pointer to class object.
+         */
+        @CEntryPoint(name = "svm_dbg_class_for_name", include = IncludeDebugHelperMethods.class, publishAs = Publish.SymbolOnly)
+        @CEntryPointOptions(prologue = SetThreadAndHeapBasePrologue.class)
+        public static Pointer getClassForName(@SuppressWarnings("unused") IsolateThread thread, Pointer ptrClassLoader, CCharPointer cClassName) {
+            Object classLoader = ptrClassLoader.toObject();
+            Log.log().string("getClassForName: ").string(cClassName).newline();
+            String className = Utf8.utf8ToString(cClassName);
+            Optional<Class<?>> opt = Heap.getHeap().getLoadedClasses().stream().filter((it) -> {
+                return it.getName().equals(className);
+            }).findFirst();
+            return Word.objectToUntrackedPointer(opt.orElse(null));
+        }
+
+        /**
+         * Allocates uninitialized instance of given class.
+         * @param thread
+         * @param ptrClass pointer to class.
+         * @return returns uninitialized instance.
+         */
+        @CEntryPoint(name = "svm_dbg_allocate_object", include = IncludeDebugHelperMethods.class, publishAs = Publish.SymbolOnly)
+        @CEntryPointOptions(prologue = SetThreadAndHeapBasePrologue.class)
+        public static Pointer getAllocateObject(@SuppressWarnings("unused") IsolateThread thread, Pointer ptrClass) {
+            Object classObject = ptrClass.toObject();
+            try {
+                return Word.objectToUntrackedPointer(Unsafe.getUnsafe().allocateInstance((Class)classObject));
+            } catch (InstantiationException e) {
+                return Word.objectToUntrackedPointer(null);
+            }
+        }
+
+        /**
+         * Pin object.
+         * @param thread
+         * @param ptrObj pointer to object to pin.
+         * @return "pin object" for given object, should be passed svm_dbg_unpin_object.
+         */
+        @CEntryPoint(name = "svm_dbg_pin_object", include = IncludeDebugHelperMethods.class, publishAs = Publish.SymbolOnly)
+        @CEntryPointOptions(prologue = SetThreadAndHeapBasePrologue.class)
+        public static Pointer pinObject(@SuppressWarnings("unused") IsolateThread thread, Pointer ptrObj) {
+            Object obj = ptrObj.toObject();
+            return Word.objectToUntrackedPointer(PinnedObject.create(obj));
```

Comment:
The `PinnedObject` itself is allocated on the Java heap and may therefore be moved by the GC.